### DISCUSSION TITLE
Term GitOps without really meaning only-git

### DISCUSSION THREAD

**Discussion by tonit** [open-gitops/project#83] [General]
This might have been a past discussion and I am just late to the game: I totally like the term GitOps but GitOps (the principles) is not tied to Git (the version control system) of cause. Even though most use-cases should be using Git at the moment.

Do we think that GitOps can live as its own term (spelling rather Gitops rather than GitOps) ?
Do we think that GitOps could confuse people not using Git - but the principles of cause would still apply? Maybe not including potential groups by sub-optimal naming?

I don't have a better term to use, just wanted to raise awareness of what i am thinking.

**Comment by bricef** [comment]
I've been struggling with this too. I definitely think the ideas we're sharing aren't tied to Git, Git just happens to be a really good example. I'm also of the mind we should standardise on "Gitops" rather than "GitOps". 

I'd entertain any useful term as a replacement. Declarative Reconciliatory State Management of Software Systems (DeReSteMaSoS) Just doesn't have the sane ring to it though. Declarative Ops? Decops? 

Sticking with GitOps/Gitops does allow us to lean on previous work, both for visibility and as reference, feels like a good-enough path of least resistance.

**Reply by tonit** [reply]
DeReSteMaSoS 😂

**Comment by schlomo** [comment]
I'd like to suggest to keep the catch phrase "GitOps" as it is very easy to remember, leans on DevOps and denotes a much higher degree of automation.

We have a good brand, let's run with it.

Everybody going in a little bit deeper soon understands that git is not that important here, and for everybody else this is "just right".

Big +1 also for leaning on previous work. The name is important and it allowed us all to rally around it in this working group here. Good names are catchy and clear even at the cost of oversimplification.

For me therefore this issue would be more about explaining the connection between git and GitOps (btw, I would keep the spelling that way): Git is the default place where most people put their desired state and it is also the common denominator for the "user interface" that people use when interfacing with their infrastructure.

**Reply by murillodigital** [reply]
I agree with keeping Git in the name, and as pointed out, as much as there may be any number of other state stores that satisfy the principles, ultimately most people will indeed simply use Git. I think the fact that GitOps is about declarative configuration and automated reconciliation doesn't make it any less important how users will build workflow to "accomplish" GitOps, all the benefits of the model become moot if it's a hassle and not operator/developer friendly, which albeit perhaps not the primary objective of GitOps, is an important aspect to the value that it brings.

**Reply by lloydchang** [reply]
Agreed.  While [Naming things is hard](https://www.karlton.org/2017/12/naming-things-hard/), it doesn't matter if you convey to the audience:
• The objective is convergence to a declarative state.

The objective is the same, while names changed over the years:
• [rdist (1983)](http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.1054.4723)
• [rsync (1996)](https://groups.google.com/g/comp.os.linux.announce/c/tZE1qtTcQaU/m/IF8GhGQ_uTsJ)
• [Heroku (2007)](https://blog.heroku.com/api_and_external_git_access)
• [GitOps (2017)](https://www.weave.works/blog/gitops-operations-by-pull-request)
• [DeReSteMaSoS (2021)](https://github.com/gitops-working-group/gitops-working-group/discussions/53#discussioncomment-363296)

Thank you @tonit @bricef @schlomo @murillodigital @o6uoq @skuethe @torstenwalter @lotharschulz @scottrigby for the conversation.

**Comment by schlomo** [comment]
In other words: I'd rather have a successful movement under the slightly incorrect term of GitOps than see the movement fail to catch momentum because we managed to find a very correct and less catchy name for it.

Like, everybody loves honey but who would eat bee-snot?

**Comment by tonit** [comment]
I totally agree with the argument pro "GitOps". 
My intend just was to clarify this here in the community and raise awareness.

Guess it should be a prominent FAQ topic on the website.

**Reply by schlomo** [reply]
To illustrate the value of the term "GitOps" I'd like to share what I got from @monadic when I asked him about it. He told me an anecdote from how he asked James Governor of RedMonk about it:

> What is your reaction, if I say the word GitOps to you? And he said, well that's possibly the ugliest word I've heard in a long time and I'm not sure I can unhear it now. So I thought, excellent, we have a winner! Because you can say this thing and it sticks in somebody's head they can't undo it, so it gives them a way, a hook into thinking about something.

And that's **exactly** the reason why I love the term GitOps 😆 

We can use this for the FAQ about why GitOps is named GitOps.

**Reply by o6uoq** [reply]
@schlomo I like that, we should pin that blurb somewhere.

**Reply by tonit** [reply]
For any future reader, here is the interview from @schlomo and @monadic in full length also covering the quote mentioned above: https://www.youtube.com/watch?v=lvLqJWOixDI
